core/src/test/java/com/coderanch/blackjack/CardsTest.java

```diff
@@ -124,4 +124,13 @@ public void getStandardDeck_returnsUnmodifiableSet() {
     public void getStandardDeck_returnsAllCombinationsOfRankAndSuit() {
         assertThat(Cards.getStandardDeck().size(), is(equalTo(Rank.values().length * Suit.values().length)));
     }
+
+    /**
+     * Tests that {@link Cards#getShuffledStandardDeck()} returns a set that contains all possible distinct cards that can be
+     * made using a combination of a {@link Rank} and a {@link Suit}.
+     */
+    @Theory
+    public void getShuffledStandardDeck_returnsAllCombinationsOfRankAndSuit() {
+        assertThat(Cards.getShuffledStandardDeck().size(), is(equalTo(Rank.values().length * Suit.values().length)));
```

Comment:
This implementation works for `getStandardDeck_returnsAllCombinationsOfRankAndSuit()` because `getStandardDeck()` returns a `Set`. As long as `getShuffledStandardDeck()` returns a `Stack` instead of a `Set` you can't be sure that it won't contain duplicates, and therefore this test method implementation is incomplete.

The easy way to fix this is to simply return a `Set` from `getShuffledStandardDeck()`.